File: src/main/java/ru/spaceinvasion/mechanic/game/Mediator.java
package ru.spaceinvasion.mechanic.game;

import org.jetbrains.annotations.Nullable;
import ru.spaceinvasion.mechanic.game.messages.GameMessage;

/**
 * Created by egor on 17.11.17.
 */

public interface Mediator<T> {
    void send(GameMessage message, T colleague, @Nullable Integer colleagueId);

    <_T extends T> void registerColleague(Class<_T> clazz, T colleague);
}

File: src/main/java/ru/spaceinvasion/mechanic/game/Mediator.kt
package ru.spaceinvasion.mechanic.game

import ru.spaceinvasion.mechanic.game.messages.GameMessage

/**
 * Created by egor on 17.11.17.
 */

interface Mediator<in T> {

    fun <_T : T> send(message: GameMessage, sendToColleague: Class<_T>, sendToColleagueId: Long?)

    fun <_T : T> registerColleague(clazz: Class<_T>, colleague: T)

}
